### ISSUE TITLE
Question about performance gap on test set

### ISSUE THREAD

**Issue by amandaluof** [IceClear/StableSR#45]
Hi! Thanks for the great work! But we found there is performance gap between ours and results reported in the paper.

**Comment by IceClear** [comment]
Did you test on the **Y** channel? It seems you tested on **RGB** channel.

**Comment by amandaluof** [comment]
> For PSNR and SSIM, did you test on the **Y** channel? It seems you tested on **RGB** channel.

Yes, we use RGB channel, you mean we need to convert all inference results and reference images to YCbCr, then calculate the metrics?

**Comment by IceClear** [comment]
> > For PSNR and SSIM, did you test on the **Y** channel? It seems you tested on **RGB** channel.
> 
> Yes, we use RGB channel, you mean we need to convert all inference results and reference images to YCbCr, then calculate the metrics?

![image](https://github.com/IceClear/StableSR/assets/22350795/bc9e824a-bb49-4703-8c48-dd3b8bc3e444)

**Comment by IceClear** [comment]
I do not think there is a big problem with your data generation code and I am sorry that I am currently on vacation and cannot access the data on the server.
First I suggest you use the same repo, i.e., IQA-pytorch for test.
Second, you can also test other baselines, e.g., RealESRGAN to see if the PSNR is also different.
Third, for real-world SR, synthetic data is just for reference, you just need to make sure the comparison is fair, i.e, test on the same data.

**Comment by ZhouJYu** [comment]
Sorry for disturbing, but as a freshman, I want to ask a question. I find that to calculate FID, we need to resize the picture to (299,299) firstly, but it is unfair to compare two HR-SR pairs with different size. It is obvious that a larger size picture will have more loss of information when resize to (299,299). I want to know whether FID score is unsuitable for evaluating large size pictures or there are some details I ignored. Hope for your reply

**Comment by amandaluof** [comment]
> Sorry for disturbing, but as a freshman, I want to ask a question. I find that to calculate FID, we need to resize the picture to (299,299) firstly, but it is unfair to compare two HR-SR pairs with different size. It is obvious that a larger size picture will have more loss of information when resize to (299,299). I want to know whether FID score is unsuitable for evaluating large size pictures or there are some details I ignored. Hope for your reply

After super resolution, the size of output images and HR images are the same.

**Comment by amandaluof** [comment]
> I do not think there is a big problem with your data generation code and I am sorry that I am currently on vacation and cannot access the data on the server. First I suggest you use the same repo, i.e., IQA-pytorch for test. Second, you can also test other baselines, e.g., RealESRGAN to see if the PSNR is also different. Third, for real-world SR, synthetic data is just for reference, you just need to make sure the comparison is fair, i.e, test all the baselines on the same data.

Thanks for your reply! To avoid the randomness in dataset making, we are trying to align other benchmark, ie. DRealSR. As I am a green bird, it is important for me to ensure my test and evaluation are both bug-free. 

I notice that you mentioned "For other methods, we directly use the ofﬁcial code and models for testing. Note that the results in this section are obtained on the same resolution with training, i.e.,128×128. Speciﬁcally, for images from [5,29,68], we crop them at the center to obtain patches with 128 × 128 resolution." 

Does this mean when evaluating RealSR and Real-ESRGAN+ on DRealSR, I need to first crop an image to a few patches with size of (128,128) ? I have test using a whole image as input, the psnr of RealSR and Real-ESRGAN+ is 30.55 and 27.92, respectively, which different from  29.75 and 30.19(gap for is RealSR acceptable)
.

**Comment by IceClear** [comment]
> > I do not think there is a big problem with your data generation code and I am sorry that I am currently on vacation and cannot access the data on the server. First I suggest you use the same repo, i.e., IQA-pytorch for test. Second, you can also test other baselines, e.g., RealESRGAN to see if the PSNR is also different. Third, for real-world SR, synthetic data is just for reference, you just need to make sure the comparison is fair, i.e, test all the baselines on the same data.
> 
> Thanks for your reply! To avoid the randomness in dataset making, we are trying to align other benchmark, ie. DRealSR. As I am a green bird, it is important for me to ensure my test and evaluation are both bug-free.
> 
> I notice that you mentioned "For other methods, we directly use the ofﬁcial code and models for testing. Note that the results in this section are obtained on the same resolution with training, i.e.,128×128. Speciﬁcally, for images from [5,29,68], we crop them at the center to obtain patches with 128 × 128 resolution."
> 
> Does this mean when evaluating RealSR and Real-ESRGAN+ on DRealSR, I need to first crop an image to a few patches with size of (128,128) ? I have test using a whole image as input, the psnr of RealSR and Real-ESRGAN+ is 30.55 and 27.92, respectively, which different from 30.19 and 29.75(gap for RealSR is acceptable) .

Yes. I think I used center crop on real-world dataset since their sizes are too large. You may have a try.
